src/libraries/Microsoft.Extensions.Primitives/src/ThrowHelper.cs

```diff
@@ -58,7 +58,19 @@ internal static ArgumentException GetArgumentException(ExceptionResource resourc
 
         private static string GetResourceText(ExceptionResource resource)
         {
-            return SR.GetResourceString(GetResourceName(resource));
+            Debug.Assert(Enum.IsDefined(typeof(ExceptionResource), resource),
+                "The enum value is not defined, please check the ExceptionResource Enum.");
+
+#pragma warning disable CS8524 // "switch is not exhaustive". It actually is.
+            return resource switch
+            {
+                ExceptionResource.Argument_InvalidOffsetLength => SR.Argument_InvalidOffsetLength,
+                ExceptionResource.Argument_InvalidOffsetLengthStringSegment => SR.Argument_InvalidOffsetLengthStringSegment,
+                ExceptionResource.Capacity_CannotChangeAfterWriteStarted => SR.Capacity_CannotChangeAfterWriteStarted,
+                ExceptionResource.Capacity_NotEnough => SR.Capacity_NotEnough,
+                ExceptionResource.Capacity_NotUsedEntirely => SR.Capacity_NotUsedEntirely,
+            };
+#pragma warning restore CS8524
```

Comment:
If an additional value is added to ExceptionResource, this will fail at runtime with a SwitchExpressionException, and since this is only triggered on failure paths where there might be catch blocks that could cause it to go unnoticed without appropriate tests in place, I think it'd be better to have an assert that'll trigger if resource is unknown.  So a pattern more like:
```suggestion
            switch (resource)
            {
                case ExceptionResource.Argument_InvalidOffsetLength: return SR.Argument_InvalidOffsetLength;
                case ExceptionResource.Argument_InvalidOffsetLengthStringSegment: return SR.Argument_InvalidOffsetLengthStringSegment;
                case ExceptionResource.Capacity_CannotChangeAfterWriteStarted: return SR.Capacity_CannotChangeAfterWriteStarted;
                case ExceptionResource.Capacity_NotEnough: return SR.Capacity_NotEnough;
                case ExceptionResource.Capacity_NotUsedEntirely: return SR.Capacity_NotUsedEntirely;
                default:
                    Debug.Fail($"Unexpected resource {resource}");
                    return "":
            };
```
or something like that.

Reply:
Fixed. I wish there was a syntax for expressing this with the `switch` expression - I feel like we keep getting pushed away from actually using the expression form due to the exhaustiveness too much.